board/embeddedartists/mx7ulpea-ucom/mx7ulpea-ucom.c

```diff
@@ -87,6 +106,264 @@ int board_early_init_f(void)
 	return 0;
 }
 
+#if defined(CONFIG_OPTEE) && defined(CONFIG_FPGA)
+/* stub TEE operations until ready to integrate with it*/
+static struct tee_hash t_hash;
+
+static int optee_get_tee_hash(uint8_t *tee_hash, int *tee_hash_len, int *payload_len)
+{
+	if (t_hash.hash.len) {
+		memcpy(tee_hash, t_hash.hash.value, t_hash.hash.len);
+		*tee_hash_len = t_hash.hash.len;
+		if (payload_len)
+			*payload_len = t_hash.payload_len;
+	}
+
+	return 0;
+}
+
+static void optee_update_tee_hash(struct hash *hash, int payload_len)
+{
+	int i;
+
+	/* stub code */
+	if (t_hash.hash.len) {
+		printf("M4: optee hash already updated, update rejected");
+		return;
+	}
+
+	t_hash.hash.len = hash->len;
+	t_hash.payload_len = payload_len;
+	memcpy(t_hash.hash.value, hash->value, t_hash.hash.len);
+
+	printf("M4: optee hash update:\n");
+	printf("\t value => ");
+	for (i = 0; i < t_hash.hash.len; i++)
+		printf("%x", t_hash.hash.value[i]);
+	printf("\n");
+	printf("\t payload : 0x%x\n", t_hash.payload_len);
+	printf("\t len     : 0x%x\n", t_hash.hash.len);
+}
+
+static int m4_do_upgrade(struct spi_flash *flash,
+			 char *data, const size_t size, struct hash *fit_hash)
+{
+	struct mtd_info *mtd = 	&flash->mtd;
+	size_t len;
+
+	/* We assume the M4 image has IVT head and padding which
+	 * should be same as the one programmed into QSPI flash
+	 */
+	if (size < 0x1000) {
+		printf("M4: Image size too small (%d)!\n", size);
+		return -EINVAL;
+	} else
+		printf("M4: Fit Image Size %d\n", size);
```

Comment:
No need for the else statement here, just move the printf line after the if block.